syntax = "proto3";

package idm;

service IDMService {
  rpc Login(LoginRequest) returns (LoginResponse);
  rpc Register(RegisterRequest) returns (RegisterResponse);
  rpc GetUser (GetUserRequest) returns (UserResponse);
  rpc UpdateUser (UpdateUserRequest) returns (UserResponse);
  rpc DeleteUser (DeleteUserRequest) returns (DeleteUserResponse);
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message LoginResponse {
  optional string token = 1;
}

message RegisterRequest {
  string username = 1;
  string email = 2;
  string password = 3;
}

message RegisterResponse {
  optional int32 id = 1;
}

message GetUserRequest {
  int32 id = 1;
}

message UpdateUserRequest {
  int32 id = 1;
  optional string email = 2;
  optional string password = 3;
  optional bool is_active = 4;
}

message UserResponse {
  int32 id = 1;
  string username = 2;
  string email = 3;
  bool is_active = 4;
  string role = 5;
}

message DeleteUserRequest {
  int32 id = 1;
}

message DeleteUserResponse {
  string message = 1;
}
